### ISSUE TITLE
Why reid is not working at all in my case?

### ISSUE THREAD

**Issue by RajS999** [mikel-brostrom/boxmot#1569]
I am trying out [boxmot](https://github.com/mikel-brostrom/boxmot) library and testing reid with botsort on small video clip in which a person walking on the road gets occluded behind a bill board and emerges out from other side of bill board. The whole clip is of mere 275 frames (~ 10 seconds). I tried several combinations of models / parameters, but the person always gets assigned with a new ID when he emerged our of bill board ! (I also tried botsort's official repo with yolov7 and yolox and reid, with no luck.) 

I tried combination of:
- yolo8n, yolo8x, 
- osnet_x0_25_msmt17.pt, clip_market1501.pt
- track buffer 30, 500

Specifically, I tried following commands:
```
/workspace/boxmot# python tracking/track.py  --yolo-model yolov8n.pt --reid-model osnet_x0_25_msmt17.pt --source <my-video-path> --save --conf 0.2
```
```
/workspace/boxmot# python tracking/track.py --yolo-model yolov8n --tracking-method botsort --reid-model clip_market1501.pt --source <my-video-path> --save --conf 0.2
```
Following with `track_buffer` = 500
```
/workspace/boxmot# python tracking/track.py --yolo-model yolov8x --tracking-method botsort --reid-model clip_market1501.pt --source <my-video-path> --save --conf 0.2
```
I believe I miss basic / essential understanding behind these algorithms? What am missing here? Why reid is not working at all and person always gets assigned with new id when he emerges out from other side of the bill board?

PS: fairmot (from official github repository) seem to work on this video clip out of the box when track_buffer size is increased.

**Comment by mikel-brostrom** [comment]
Once a track is marked as "lost" due to an unsuccessful association (i.e., when no detections match the track), the Kalman Filter (KF) associated with that track stops updating its state. This happens after a single unsuccessfull  association.

**Comment by RajS999** [comment]
Am trying to make reid work with botsort. Thats why I tried botsort's official repo and thats why I specified `--tracking-method botsort` in second and third command. But mistakenly I did not specify it in first command, so the first command is basically invalid in the sense that it uses the default `deepocsort`. Sorry I did not explicitly specify this earlier ! Now I guess, botsort does not have `max_age` parameter, right? In that case, what might be the reason for reid to not work with it? I also tried with `proximity_thresh: 0, track_buffer: 500` in botsort.yaml.

**Comment by RajS999** [comment]
I re-read through [BoT-SORT paper](https://arxiv.org/pdf/2206.14651) and [Fast-ReID paper](https://arxiv.org/pdf/2006.02631). These are the excerpts from these papers:

- From Fast ReID paper: We propose a cross-domain method FastReIDMLT that adopts mixture label transport to learn pseudo label by multi-granularity strategy. We first train a model with a source-domain dataset and then finetune on the pre-trained model with pseudo labels of the target-domain dataset. Even after fine tuning, it achieved poorer performance for some source-target datasets for cross domain person re-identification. (Even after fine tuning the authors noted down poorer performance for some combination of source - target datasets for cross domain person re-identification as can be seen in second last line in below table.)
![image](https://github.com/user-attachments/assets/93c5c970-2674-4e9d-a456-6bb3bda76d92)

- From BoT-SORT paper: For the feature extractor, we trained FastReID’s SBS-50 model for MOT17 and MOT20 with their default training strategy, for 60 epochs. While we trained on the first half of each sequence and tested on the rest.

Now I have two doubts:

**Q1.** Does above mean we cannot simply use out of the box Re-ID models for successfully doing cross-domain person reidentification? We need to fine-tune them on our dataset before using?

**Q2.** What are SOTA or community preferred methods for battery powered on-device object tracking with re-identification?
